### PR TITLE
migrated web_dashboard_tile from version 10 to 11

### PR DESCRIPTION
**Pull request by steigendit**
migrated web_dashboard_tile from version 10 to 11

### CODE REVIEW

**Comment by tarteo** [comment]
Can you include the previous commits. How to do it: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0#technical-method-to-migrate-a-module-from-100-to-110-branch

It's easier to review that way.

File: web_dashboard_tile/__init__.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# -*- coding: utf-8 -*-
+# © 2010-2013 OpenERP s.a. (<http://openerp.com>).
```
Comment (tarteo) [inline web_dashboard_tile/__init__.py:2]: Replace the copyright character by 'Copyright'

File: web_dashboard_tile/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        'board',
+        'mail',
+        'web_widget_color',
+    ],
+    'author': 'initOS GmbH & Co. KG, '
+              'GRAP, '
+              'Odoo Community Association (OCA)',
+    "category": "web",
+    'license': 'AGPL-3',
+    'contributors': [
```
Comment (tarteo) [inline web_dashboard_tile/__manifest__.py:20]: I don't think this works. You can use 'maintainers' though https://github.com/OCA/maintainer-tools/blob/771508e99876115c418888bbac7899174662482b/template/module/__manifest__.py#L14

File: web_dashboard_tile/data/res_groups.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+<odoo>
+    <data>
+    <!--  !record {model: res.groups, id: base.group_no_one}:
+    users:
+        - base.user_root -->
+        <record id="base.group_no_one" model="res.groups">
+            <field name="users" eval="[(4, ref('base.user_root'))]"/>
+        </record>
+    </data>
+</odoo>
```
Comment (tarteo) [inline web_dashboard_tile/data/res_groups.xml:11]: Missing new line

File: web_dashboard_tile/static/src/js/custom_js.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    _t =  core._t,
+   _lt = core_lt;
+_.mixin({
+  sum: function (obj) { return _.reduce(obj, function (a, b) { return a + b; }, 0); }
+});
+    var module = instance.board.AddToDashboard;
+
+    module.include({
+        start: function () {
+        	console.log("dashboard tileeeeeeeeeeeeeeeeeeee starttttttttt")
```
Comment (tarteo) [inline web_dashboard_tile/static/src/js/custom_js.js:36]: Uhm, haha :smile:

File: web_dashboard_tile/static/src/js/custom_js.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
+//  @@@ web_dashboard_tile custom JS @@@
```
Comment (tarteo) [inline web_dashboard_tile/static/src/js/custom_js.js:1]: Can you check jslinter for errors, please.

File: web_dashboard_tile/static/src/js/custom_js.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+            _.each(search.contexts, context.add, context);
+            _.each(search.domains, domain.add, domain);
+
+            var c = instance.web.pyeval.eval('context', context);
+            for(var k in c) {
+                if (c.hasOwnProperty(k) && /^search_default_/.test(k)) {
+                    delete c[k];
+                }
+            }
+            // TODO: replace this 6.1 workaround by attribute on <action/>
```
Comment (tarteo) [inline web_dashboard_tile/static/src/js/custom_js.js:74]: Is the workaround still needed?

**Review by tarteo — CHANGES_REQUESTED** [review]
Thanks!

File: web_dashboard_tile/__init__.py (hunk 1 above)
Comment (omalbastin) [reply web_dashboard_tile/__init__.py:2]: Contributing modules for the first time. We will make the necessary changes and update. Thanks @tarteo

File: web_dashboard_tile/models/tile_tile.py [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+from odoo import api, fields, models
+from odoo.tools.safe_eval import safe_eval as eval
+from odoo.tools.translate import _
+from odoo.exceptions import ValidationError, except_orm
+
+
+def median(vals):
+    # https://docs.python.org/3/library/statistics.html#statistics.median
+    # TODO : refactor, using statistics.median when Odoo will be available
+    #  in Python 3.4
```
Comment (daramousk) [inline web_dashboard_tile/models/tile_tile.py:21]: The version of Odoo we migrate this module to is already on python `>=3.5` so remove this function and replace it with `statistics.median`

File: web_dashboard_tile/models/tile_tile.py [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+        compute='_compute_data')
+    secondary_helper = fields.Char(
+        string='Secondary Helper',
+        compute='_compute_helper')
+
+    error = fields.Char(
+        string='Error Details',
+        compute='_compute_data')
+
+    @api.one
```
Comment (daramousk) [inline web_dashboard_tile/models/tile_tile.py:145]: Remove api.one from everywhere and replace with api.multi and deal with the fact that `self` might contain more than one records.

File: web_dashboard_tile/static/src/js/custom_js.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+//    This program is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+//#############################################################################
+odoo.define('web_dashboard_tile', function (require) {
```
Comment (daramousk) [inline web_dashboard_tile/static/src/js/custom_js.js:21]: Please fix stuff in here, licence, uneeded comments/workarounds, log statements,formatting etc

**Review by daramousk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**